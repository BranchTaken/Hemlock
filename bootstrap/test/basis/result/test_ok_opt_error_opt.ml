open! Basis.Rudiments
open! Basis
open Result
open Format

let test () =
  printf "@[<h>";
  List.iter [Ok "ok"; Error "error"] ~f:(fun result ->
    printf "ok_opt %a -> %a\n"
      (pp String.pp String.pp) result
      (Option.pp String.pp) (ok_opt result);
    printf "error_opt %a -> %a\n"
      (pp String.pp String.pp) result
      (Option.pp String.pp) (error_opt result);
  );
  printf "@]"

let _ = test ()
