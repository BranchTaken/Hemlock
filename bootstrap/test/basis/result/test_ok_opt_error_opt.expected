ok_opt Ok ("ok") -> Some "ok"
error_opt Ok ("ok") -> None
ok_opt Error ("error") -> None
error_opt Error ("error") -> Some "error"
